### PR TITLE
Add https support and authentication to marathon plugins

### PR DESCRIPTION
**Pull request by marcdeop**
## Pull Request Checklist

This PR is not referring to an existing issue.
#### General
- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)
- [x] RuboCop passes
- [x] Existing tests pass 
#### Purpose

Add https support and authentication to marathon plugins:
- Add "protocol" option to check-marathon and metrics-marathon
- Add "protocol", "username" and "password" options to check-marathon-task

### CODE REVIEW

File: bin/check-marathon-task.rb [hunk 1]
```diff
@@ -50,20 +50,35 @@ class MarathonTaskCheck < Sensu::Plugin::Check::CLI
...
 
   def run
-    if config[:instances] == 0
+    if config[:instances].zero?
       unknown 'number of instances should be an integer'
     end
 
+    if !config[:username].nil? && config[:password].nil? ||
+       config[:username].nil? && !config[:password].nil?
+      unknown 'You must provide both username and password'
```
Comment (majormoses) [inline bin/check-marathon-task.rb]: should consider adding an option to allow user to crit on failure?

**Comment by majormoses** [comment]
@marcdeop just a few comments, could you address them?

File: bin/check-marathon-task.rb (hunk 1 above)
Comment (marcdeop) [inline bin/check-marathon-task.rb]: Isn't that out of the scope of the pull request? I mean... that has nothing to do with adding support for https and authentication, does it?

We could create a new "Issue" so we can add that option and keep things isolated. How about that?

Comment (majormoses) [inline bin/check-marathon-task.rb]: I would not say so personally as you are adding new functionality and I am suggesting how that feature should work. That being said it was a suggestion and not a requirement.

**Comment by majormoses** [comment]
@marcdeop we can drop ruby 1.9 testing as we no longer support it. this should allow tests to pass if not I can do that for you.

**Comment by majormoses** [comment]
@marcdeop merged, open an issue and ping me if I do not release within 24 hours.

**Pull request merged**